mautic/mautic
# Force to array for groupBy and aggregators in reports

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | features
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | yes
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 




#### Description:
This happend If you use default reports from older instances. If aggregators and orderBy columns are null it return a lot of errors for Leads and Points report (id 5 after installation)

![image](https://user-images.githubusercontent.com/462477/98700457-7e7b7580-2378-11eb-8b2c-6f889e138f09.png)

Errors:

`[2020-10-26 16:15:40] mautic.WARNING: PHP Warning - array_merge(): Argument #2 is not an array - in file /path/app/bundles/ReportBundle/Entity/Report.php - at line 505 [] [] [2020-10-26 16:15:40] mautic.WARNING: PHP Warning - array_merge(): Argument #1 is not an array - in file /path/app/bundles/ReportBundle/Entity/Report.php - at line 497 [] [] [2020-10-26 16:15:40] mautic.WARNING: PHP Warning - in_array() expects parameter 2 to be array, null given - in file /path/app/bundles/ReportBundle/Event/ReportGeneratorEvent.php - at line 400 {"column":"comp.companyname","columns":null} []`



#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create basic report
3. Force set aggregators and group_by columns to null 
4. See errors in report details reported above
5. After fix it should work properly
